src/Event/DoiDatasetListener.php

```diff
@@ -19,7 +21,7 @@ public function onDeleteDoi(EntityEvent $event)
         $dataset = $event->getEntity();
         $doi = $dataset->getDoi();
         if ($doi) {
-            $this->producer->publish($doi->getDoi(), 'delete');
+            $this->publisher->publish($doi->getDoi(), RabbitPublisher::DOI_PRODUCER, 'delete');
```

Comment:
Need to change RabbitPublisher's publish method to accept type string argument also, because dataset id is deleted by the API before the event is triggered. Should refactor the way we delete Dois on Dataset deletion in the future